### PR TITLE
Remove KVM from PMO

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR removes all KVM related things to clean up PMO a bit because we have not been supporting KVM in a while in PMO anyway

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: flag/service/prometheus/prometheus.go [hunk 1]
```diff
@@ -8,7 +8,6 @@ type Prometheus struct {
 	EvaluationInterval      string
 	ImageRepository         string
 	LogLevel                string
-	Mayu                    string
```
Comment (QuentinBisson) [inline flag/service/prometheus/prometheus.go:11]: Mayu is a KVM only target

File: helm/prometheus-meta-operator/templates/rbac.yaml [hunk 2]
```diff
@@ -93,24 +93,6 @@ rules:
       - create
       - delete
       - update
-  - apiGroups:
```
Comment (QuentinBisson) [inline helm/prometheus-meta-operator/templates/rbac.yaml:96]: By removing the legacy controller only used on KVM, we are not using AwsConfig and AzureConfigs

File: pkg/unittest/input/case-1-awsconfig.golden [hunk 3]
```diff
@@ -1,8 +1,13 @@
-apiVersion: provider.giantswarm.io/v1alpha1
-kind: AWSConfig
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
 metadata:
```
Comment (QuentinBisson) [inline pkg/unittest/input/case-1-awsconfig.golden:3]: Will rename the files in a subsequent PR

**Review by QuantumEnigmaa — APPROVED** [review]

**Comment by QuentinBisson** [comment]
@marieroque and @hervenicol could i have another pair of eyes just to be safe? Thank you

**Comment by marieroque** [comment]
LGTM if tested on CAPI and Vintage :-)

**Review by marieroque — APPROVED** [review]

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**